test/OpenTelemetry.Extensions.PersistentStorage.Tests/FileBlobProviderTests.cs

```diff
@@ -16,143 +16,146 @@
 
             // Verify file name from both create blob and get blob are same.
-            Assert.Equal(((FileBlob)blob1).FullPath, ((FileBlob)blob2).FullPath);
+            Assert.Equal(((FileBlob)blob).FullPath, ((FileBlob)singleBlob).FullPath);
 
             // Validate if content in the blob is same as buffer data passed to create blob.
-            Assert.Equal(data, blob1.Read());
+            Assert.True(blob.TryRead(out var blobContent));
+            Assert.Equal(data, blobContent);
 
             testDirectory.Delete(true);
         }
 
         [Fact]
-        public void FileStorage_CreateBlobReturnsNullIfStorageIsFull()
+        public void FileBlobProvider_CreateBlobReturnsNullIfblobProviderIsFull()
         {
             var testDirectory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
-            using var storage = new FileStorage(testDirectory.FullName, 10000);
+            using var blobProvider = new FileBlobProvider(testDirectory.FullName, 10000);
 
             PersistentStorageHelper.UpdateDirectorySize(10000);
 
             var data = Encoding.UTF8.GetBytes("Hello, World!");
 
-            Assert.Null(storage.CreateBlob(data));
+            Assert.False(blobProvider.TryCreateBlob(data, out var blob));
+            Assert.Null(blob);
 
             testDirectory.Delete(true);
         }
 
         [Fact]
-        public void FileStorage_PathIsRequired()
+        public void FileBlobProvider_PathIsRequired()
         {
-            Assert.Throws<ArgumentNullException>(() => new FileStorage(null));
+            Assert.Throws<ArgumentNullException>(() => new FileBlobProvider(null));
         }
 
         [Fact]
-        public void FileStorage_TestRetentionPeriod()
+        public void FileBlobProvider_TestRetentionPeriod()
         {
             var testDirectory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
             long maxSizeInBytes = 100000;
             int maintenancePeriodInMilliseconds = 3000;
             int retentionPeriodInMilliseconds = 2000;
             int writeTimeOutInMilliseconds = 1000;
-            using var storage = new FileStorage(
+            using var blobProvider = new FileBlobProvider(
                 testDirectory.FullName,
                 maxSizeInBytes,
                 maintenancePeriodInMilliseconds,
                 retentionPeriodInMilliseconds,
                 writeTimeOutInMilliseconds);
 
             var data = Encoding.UTF8.GetBytes("Hello, World!");
-            var blob1 = (FileBlob)storage.CreateBlob(data);
+            Assert.True(blobProvider.TryCreateBlob(data, out var blob));
 
             // Wait for maintenance job to run
             Thread.Sleep(4000);
 
             // Blob will be deleted as retention period is 1 sec
-            Assert.False(File.Exists(blob1.FullPath));
+            Assert.False(File.Exists(((FileBlob)blob).FullPath));
 
             testDirectory.Delete(true);
```

Comment:
Consider putting the `Delete()` in a `finally`. If any of the above asserts fail, I think this won't be executed.

Reply:
I think, we dont need to do that. I will do a follow up PR and remove it.